### PR TITLE
Domains: Update Domain Transfer checkout page

### PR DESCRIPTION
**Pull request by rafaelgallani**
#### Changes proposed in this Pull Request
This PR implements the changes on the domain transfer checkout page layout. 
Here's a comparison of how they looked before vs. how they're looking now: 

##### Before
before - desktop

##### Before
before - mobile

#### Testing instructions
 - Add a domain transfer to the cart;
 - Checkout and confirm that the new "thank you" page looks like this. 

_(If you already have a domain transferred and you have the receipt ID, you can bypass the checkout by going to `/checkout/thank-you/:your-site/:receipt-id` 😉)_

### CODE REVIEW

**Review by poligilad-auto — CHANGES_REQUESTED** [review]
Looks good @rafaelgalani. On mobile the only thing I would change is making sure the area with the black strip has the same padding as the content area below it. So then if the text is too long it will go to the next line and keep the icon aligned to the first line of text, like a bullet point.

image

Also you mentioned that the additions I made to the designs will go in a separate PR?

**Comment by rafaelgallani** [comment]
@poligilad-auto OK, adding the changes 😄 

> Also you mentioned that the additions I made to the designs will go in a separate PR?

No. 😸 
Since they're simple enough (only a few sections), they can go into this one. 

I just added the new sections and updated the layout. Let me know if that looks good 😄

**Comment by rafaelgallani** [comment]
Rebasing on `trunk`.

**Comment by rafaelgallani** [comment]
@poligilad-auto Here's how it looks now: 

#### Desktop
image

#### Mobile
image

**Review by poligilad-auto — APPROVED** [review]

File: client/my-sites/checkout/checkout-thank-you/domains/style.scss [hunk 1]
```diff
@@ -117,5 +121,13 @@
 				}
 			}
 		}
+
+		&__notice {
+			@include breakpoint-deprecated( '<480px' ) {
+				padding: 14px 24px;
+    			align-items: flex-start;
+    			height: auto;
```
Comment (delputnam) [inline client/my-sites/checkout/checkout-thank-you/domains/style.scss]: Super minor nit: indention spacing.

**Review by delputnam — APPROVED** [review]
Really good job on this one.  Thanks for sticking with it!

File: client/my-sites/checkout/checkout-thank-you/domains/style.scss (hunk 1 above)
Comment (rafaelgallani) [reply client/my-sites/checkout/checkout-thank-you/domains/style.scss]: Totally missed that... IDK why that happened here.. 🤨 
image

But thanks!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6576301

Thank you @rafaelgalani for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6576301).

**Pull request merged**